includes/widgets/class-wc-widget-product-categories.php

```diff
@@ -245,8 +249,27 @@ public function widget( $args, $instance ) {
 							this_page = home_url + '?product_cat=' + jQuery(this).val();
 						}
 						location.href = this_page;
+					} else {
+						location.href = '" . esc_js( wc_get_page_permalink( 'shop' ) ) . "';
```

Comment:
How do you know this is on the shop page?

Needs to remove product_cat from current URl.

Reply:
It is the same way layer nav does it. When viewing categories it is technically the shop page, when you just remove product_cat it will redirect to the home page always, which might not contain any products that is why shop page makes sense, if you are busy viewing categories and clear the filter I expect it to show all product.